autoload/coverage/vim.vim

```diff
@@ -0,0 +1,28 @@
+" Copyright 2017 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+"{{{ covimerage provider (based on coverage.py)
+
+function! coverage#vim#GetCovimerageProvider() abort
+  let l:provider = extend(coverage#python#GetCoveragePyProvider(), {
```

Comment:
I think in practice GetCoveragePyProvider generates a fresh dict on each call anyway, but could you `copy()` before extending just so it's easy to visually verify there are no subtle side effects to this code?

Reply:
Done (amended).

Reply:
Should be `deepcopy` then though after all, shouldn't it?

Reply:
Could be. It'd be a lot less likely to accidentally mutate one level down though, and none of the contained values are lists or dicts that could be modified in place anyway.